EXAMPLES.md

```diff
@@ -0,0 +1,202 @@
+# Examples
+
+- [Request Options](#request-options)
+- [Pagination](#pagination)
+  - [Page based pagination](#page-based-pagination)
+  - [Checkpoint pagination](#checkpoint-pagination)
+- [Custom User Structs](#providing-a-custom-user-struct)
+
+## Request Options
+
+Fine-grained configuration can be provided on a per-request basis to enhance the request with specific query params, headers or to pass it a custom context.
+
+```go
+// Example
+userGrants, err := auth0API.Grant.List(
+	management.Context(ctx)
+	management.Header("MySpecialHeader","MySpecialHeaderValue")
+    management.Parameter("user_id", "someUserID"),
+    management.Parameter("client", "someClientID"),
+    management.Parameter("audience", "someAPIAudience"),
+)
+
+// Other helper funcs.
+management.Query()
+management.ExcludeFields()
+management.IncludeFields()
+management.Page()
+management.PerPage()
+management.IncludeTotals()
+management.Take()
+management.From()
+```
+
+## Pagination
+
+This SDK supports both offset and checkpoint pagination.
+
+### Page based pagination
+
+When retrieving lists of resources, if no query parameters are set using the `management.PerPage` and `Management.IncludeTotals` helper funcs, then the SDK will default to sending `per_page=50` and `include_totals=true`. 
+
+> **Note**
+> The maximum value of the per_page query parameter is 100.
+
+In order to paginate using the page based pagination, you can follow a pattern like below:
+
+```go
+var page int
+for {
+    clients, err := auth0API.Client.List(
+        management.Page(page),
+        management.PerPage(100),
+    )
+    if err != nil {
+        return err
+    }
+
+    // Accumulate here the results or check for a specific client.
+
+    if !clients.HasNext() {
+        break
+    }
+
+    page++
+}
+```
+
+### Checkpoint pagination
+
+Checkpoint pagination can be used when you wish to retrieve more than 1000 results from certain APIs. The APIs that support checkpoint based pagination are:
+
+* `Log.List` (`/api/v2/logs`)
+* `Organization.List` (`/api/v2/organizations`)
+* `Organization.Members` (`/api/v2/organizations/{id}/members`)
+* `Role.Users` (`/api/v2/roles/{id}/users`)
+
+In general to use checkpoint based pagination you should follow a pattern like below:
+
+```go
+var orgList *management.OrganizationList
+for {
+    if orgList == nil {
+        // For the first call, only pass the `take` query parameter, the API will then return a
+        // `Next` value that can be used for future requests,
```

Comment:
```suggestion
        // `Next` value that can be used for future requests.
```

Reply:
Resolved as part of the change to this code (and the others below it)